### PR TITLE
Add automated script to verify contents of publish.yml and publish-appservice-stage.yml

### LINKED ISSUE #480: Add test to ensure staged publish includes images listed in publish
Add test that verifies - images in - https://github.com/Azure/azure-functions-docker/blob/dev/host/3.0/publish.yml are included in https://github.com/Azure/azure-functions-docker/blob/dev/host/3.0/publish-appservice-stage.yml

### PR DESCRIPTION
**Pull request by CooperLink**
### PR information
Added script that ensure that all -appservice images in the publish-appservice-stage.yml pipeline also exist in the publish.yml pipeline. And vice versa. There is an exception for the :base-VERSION-appservice image as that is published to public MCR but, not the staging regions. 

This PR will block the automatedPR github action unless both pipelines are consistent.  This will aid us in the future by ensuring that we are always publishing all images we intend to in both the stage and public environment. 

Fixes #480 
Fix for work item here : https://github.com/Azure/azure-functions-docker/issues/480#issuecomment-908513317

Example : Node 8 images were added in May to the public MCR publishing pipeline but, did not get added to the -stage pipeline until August 11th.  This is an easy mistake to make, so this script will help us avoid doing it again :) 

https://github.com/Azure/azure-functions-docker/commit/770627271370384311d7a32eb73f5eae093a51d4#diff-6ea5486dcbbf0a025e064169297b8989a4491f26366213bddf6e8b4e18b1ca34

https://github.com/Azure/azure-functions-docker/commit/ec88dea8fc0370c04404c9409dfae17de9f7d4e6#diff-3159601f24e38c999bc9ebb369d1dcb81e5b03d457c65c4388de3fdb45962238

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/development-docs/cleaning-up-commits.md).
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### CODE REVIEW

**Comment by pragnagopa** [comment]
tagging @sidkri  as FYI!

**Review by ahmelsayed — APPROVED** [review]

**Review by pragnagopa — APPROVED** [review]
💯

File: .github/workflows/autopr.yaml [hunk 1]
```diff
@@ -22,6 +22,8 @@ jobs:
       run: |
         git config --local user.email "action@github.com"
         git config --local user.name "GitHub Action"
+        cd host/3.0
+        ./verify-pipelines.sh
```
Comment (sidkri) [inline .github/workflows/autopr.yaml:26]: Please verify this in a test pipeline run as I'm not sure that the following commands will not be run based on the exit status, you may need to check the exit status and exit on this line.

Comment (CooperLink) [reply .github/workflows/autopr.yaml:26]: Thank you for your suggestion. Successfully verified the pipeline behavior.  Sent you an image offline but also feel free to checkout this workflow
https://github.com/Azure/azure-functions-docker/actions/runs/1242979688

**Comment by balag0** [comment]
@CooperLink In addition to this, I think we also need to extend this new validation v4. Not urgent yet, but could you open a task to track please.

May be we create a separate action for v4? + @TsuyoshiUshio as well

**Comment by pragnagopa** [comment]
Thanks @balag0 for the ping here. Yes we should extend verification to include V4 publish flows as well. @CooperLink - please track the update in a separate PR

**Pull request merged**